src/explorer.api/Exploration/ExplorationRegistry.cs

```diff
@@ -20,26 +27,19 @@ public Guid Register(Exploration exploration, CancellationTokenSource tokenSourc
             return id;
         }
 
-        public async Task Remove(Guid id)
+        public void Remove(Guid id)
         {
-            registrations.Remove(id, out var registration);
-
-            try
+            if (!registrations.TryGetValue(id, out var registration))
             {
-                if (registration is null)
-                {
-                    return;
-                }
-                registration.CancelExploration();
-
-                // Wait for the task to finish before disposing the registration.
-                // This will also allow any Exceptions to surface.
-                await registration.Exploration.Completion;
+                return;
             }
-            finally
+
+            if (registration.Status == ExplorationStatus.Processing)
```

Comment:
I believe that the condition should also check check the `ExplorationStatus.New`.

Reply:
My thinking was that if it's `New` then it hasn't started running, so it doesn't need to be canceled. 

But actually I think you're right - if the exploration isn't already completed (either canceled or errored or completed successfully) then it shouldn't be removed and we should throw an error.

Reply:
Done
c8a0966